crates/sui-graphql-rpc/src/types/checkpoint.rs

```diff
@@ -15,8 +19,52 @@ pub(crate) struct Checkpoint {
     pub live_object_set_digest: Option<String>,
     pub network_total_transactions: Option<u64>,
     pub rolling_gas_summary: Option<GasCostSummary>,
-    pub epoch: Option<Epoch>,
+    pub epoch: Epoch,
     pub end_of_epoch: Option<EndOfEpochData>,
     // transactionConnection(first: Int, after: String, last: Int, before: String): TransactionBlockConnection
     // address_metrics: AddressMetrics,
 }
+
+impl From<&sui_json_rpc_types::Checkpoint> for Checkpoint {
+    fn from(c: &sui_json_rpc_types::Checkpoint) -> Self {
+        let end_of_epoch_data = &c.end_of_epoch_data;
+        let end_of_epoch = end_of_epoch_data.clone().map(|e| {
+            let committees = e.next_epoch_committee;
+            let new_committee = if committees.is_empty() {
+                None
+            } else {
+                Some(
+                    committees
+                        .iter()
+                        .map(|c| CommitteeMember {
+                            authority_name: Some(c.0.into_concise().to_string()),
+                            stake_unit: Some(c.1),
+                        })
+                        .collect::<Vec<_>>(),
+                )
+            };
+
+            EndOfEpochData {
+                new_committee,
+                next_protocol_version: Some(e.next_epoch_protocol_version.as_u64()),
+            }
```

Comment:
nit: slightly neater derivation in suggestion, but also it's a little strange to me that we are translating an empty vector into `None` -- why have we wrapped this type in `Option` if we treat `None` and `Some(vec![])` the same?

```suggestion
            let committee: Vec<_> = e
                .next_epoch_committee
                .iter()
                .map(|(authority, stake)| CommitteeMember {
                    authority_name: Some(authority.into_concise().to_string()),
                    stake_unit: Some(stake),
                })
                .collect();

            EndOfEpochData {
                new_committee: (!commitee.is_empty()).then_some(committee),
                next_protocol_version: Some(e.next_epoch_protocol_version.as_u64()),
            }
```

Reply:
I'll be honest with you, I can't answer your question because I copied the existing code over :painlaugh:

Reply:
But yeah agreed, looks like EndOfEpochData.newCommittee is an optional vec of CommitteeMembers, makes sense to me in that Some([]) should be a valid value